testflinger_agent/client.py

```diff
@@ -81,8 +81,14 @@ def _configure_influx(self):
             influx_client.ping()
         except ConnectionError as exc:
             logger.error(exc)
-        else:
-            return influx_client
+            return None
+
+        # create agent job db if it doesn't exist
+        db_list = influx_client.get_list_database()
```

Comment:
Oh gosh, yes! we definitely need this. It was definitely in the review I did of https://github.com/canonical/testflinger-agent/pull/29 also, why was it marked resolved if it wasn't actually resolved?

The only thing I have to say about this, is please check whether it's even necessary to check that the database exists. I *think* that it might be ok to create it even if it already exists, but it's good to test that first.

Reply:
Yeah, it's strange; I think in creating and refactoring the helper method to config influx I accidentally dropped that part of it.

I think we can refactor the db check to just duck type around the db creation, I'll look into that today.